
### SAVER-X: single-cell analysis via expression recovery harnessing external data {#ch-5-1-2}

SAVER-X [@RN204] (please check this)<!--[56???? Is this SAVER-X Ref?]--> is an AE model developed to denoise and impute scRNA-seq data with transfer learning from other data resources. SAVER-X is implemented in R with the support of Python package sctransfer. 
Model. SAVER-X decomposes the variation in the observed counts $x_{n}$ with missing values into three components: i predictable structured component that represents the shared variation across genes, ii) unpredictable cell-level biological variation and gene-specific dispersions, and iii) technical noise. Specifically,  $x_{gn}$ is modeled as a Poisson-Gamma hierarchical model,

\begin{equation}
p(x_{gn}\vert \Theta)=Poisson(l_{n}x'_{gn}), \space \space p(x'_{gn}\vert v_{gn},\alpha_{g})=Gamma(v_{gn},\alpha_{g}v_{gn}^{2}) (\#eq:eq14)
\end{equation}

where $l_{n}$ is the sequencing depth of cell $n$, $ν_{gn}$ is the mean, and $\alpha_{g}$ is the dispersion. This Poisson-Gamma mixture is an equivalent expression to the NB distribution.  As a result, the ZINB distribution as Eq. (\@ref(eq:eq11)) in DCA is also adopted to model missing values. 
To train the model, a similar log-likelihood function as Eq. (\@ref(eq:eq13)) in DCA is used as the loss function. However, $ν_{gn}$ is initially learned by an AE pre-trained using external datasets from an identical or similar tissue and then transferred to and updated using data $x_{n}$ to be denoised. Such transfer learning can be applied to data between species (e.g., human and mouse in the study), cell types, batches, and single-cell profiling technologies. After $ν_{gn}$ is inferred, SAVER-X generates the final denoised data $\hat{x}_{gn}$ by an empirical Bayesian shrinkage.  

*Evaluation metrics.*  $t$-SNE visualization and ARI were used to evaluate the clustering performance after imputation. Pearson correlation coefficients between protein and corresponding gene expression levels after denoising CITE-seq data were computed. 

*Results.* SAVER-X was applied to multiple human single-cell datasets of different scenarios: i) T-cell subtypes, ii) a cell type (CD4+ tegulatory T cells) that was absent from the pretraining dataset, iii) gene-protein correlations of CITE-seq data, and iv) immune cells of primary breast cancer samples with a pretraining on normal immune cells. SAVER-X with pretraining on HCA and/or PBMCs outperformed the same model without pretraining and other denoising methods, including DCA [@RN80]<!--[23]-->, scVI [@RN88]<!--[13]-->, scImpute [@RN31]<!--[33]-->, and MAGIC [@RN111]<!--[55]-->. The model achieved promising results even for genes with very low UMI counts. SAVER-X was also applied for a cross-species study in which the model was pretrained on a human or mouse dataset and transferred to denoise another. The results demonstrated the merit of transferring public data resources to denoise in-house scRNA-seq data even when the study species, cell types, or single-cell profiling technologies are different.

















