
### scGen, a generative model to predict perturbation response of single cells across cell types {#ch-5-7-2}

scGen [@RN179]<!--[122]--> is designed to learn cell response to certain perturbation (drug treatment, gene knockout, etc) from single cell expression data and predict the response to the same perturbation for a new sample or a new cell type. The novelty of scGen is that it learns the response in the latent space instead of the expression data space.

*Model.* ScGen follows the general VAE for scRNA-seq data but uses the “latent space arithmetics” to learn perturbations' response. Given scRNAseq samples of perturbed (denoted as $p$) and unperturbed cells (denoted as $unp$), a VAE model is trained.  Then, the latent space representation $z_{p}$ and $z_{unpare}$ obtained for the perturbed and unperturbed cells. Following the notion that VAE could map nonlinear operations (e.g., perturbation) in the data space to linear operations in the latent space, ScGen estimate the response in the latent space as  $\delta=\bar{z}_{p}-\bar{z}_{unp}$, where $\bar{z}$ is the average representation of samples from the same cell type or different cell types. Then,  given the latent representation $z'_{unp}$ of an unperturbed cell for a new sample from the same or different cell type, the latent representation of the corresponding perturbed cell can be predicted as $z'_{p}=z'_{unp}+\delta$. The expression of the perturbed cell can also be estimated by feeding $z'_{p}$ into the VAE decoder. The scGen can also be expanded to samples and treatment across two species (using orthologues between species). When scGen is trained for species 1, or $s1$, with both perturbed and unperturbed cells but species 2, $s2$, with only unperturbed cells, then the latent code for the perturbed cells from $s2$ can be predicted as $z_{s_{2},p} = \frac{1}{2}(z_{s_{1},p} + z_{s_{2},unp} + \delta_{s} + \delta_{p})$ where $\delta_{p} = z_{s_{1},unp} - z_{s_{1},p}$ captures the response of perturbation and $\delta_{s} = z_{s_{1}}-z_{s_{2}}$ represents the difference between species. 

*Evaluation matrics.*  Correlation and UMAP visualization were used to assess the performance. 

*Result.* scGen was applied to predict perturbation of out-of-samples response in human PBMCs data, and scGen showed a higher average correlation ($R^{2}$= 0.948) between predicted and real data for six cell types. Comparing with other methods including CVAE [@RN201]<!--[123]-->, style transfer GAN [*124:reference not found*]<!--[124]-->, linear approaches based on vector arithmetics (VA) [@RN179]<!--[122]--> and PCA+VA, scGen predicted full distribution of ISG15 gene (strongest regulated gene by IFN-$\beta$) response to IFN-$\beta$ [@RN218]<!--[125]-->, while others might predicted mean (CAVE and style transfer GAN) but failed to produce the full distribution. scGen was also tested on predicting the intestinal epithelial cells’ response to infection [@RN226]<!--[126]-->. For early transit-amplifying cells, scGen showed good prediction ($R^{2}$=0.98) for both H. poly and Salmonella infection.  Finally, scGen was evaluated for perturbation across species using scRNA-seq data set by Hagai et al [@RN202]<!--[127]-->, which comprised of bone marrow-derived mononuclear phagocytes from mice, rats, rabbits, and pigs perturbed with lipopolysaccharide (LPS). scGen’s predictions of LPS perturbion responses were shown to be highly correlated ($R^{2}$ = 0.91) with the real responses.




