# [Deep learning tackles single-cell analysis – A survey of deep learning for scRNA-seq analysis](https://huang-ai4medicine-lab.github.io/survey-of-DL-for-scRNA-seq-analysis/_book)

**Mario Flores<sup>1§</sup>, Zhentao Liu<sup>1</sup>, Tinghe Zhang<sup>1</sup>, Md Musaddaqui Hasib<sup>1</sup>, Yu-Chiao Chiu<sup>2</sup>, Zhenqing Ye<sup>2,3</sup>, Karla Paniagua<sup>1</sup>, Sumin Jo<sup>1</sup>, Jianqiu Zhang<sup>1</sup>, Shou-Jiang Gao<sup>4,6</sup>, Yufang Jin<sup>1</sup>, Yidong Chen<sup>2,3§</sup>, and Yufei Huang<sup>5,6§</sup>**

<sup>§</sup>Corresponding authors: Mario Flores (mario.flores@utsa.edu); Yidong Chen (cheny8@uthscsa.edu); Yufei Huang (yuh119@pitt.edu)

<sup>1</sup>Department of Electrical and Computer Engineering, the University of Texas at San Antonio, San Antonio, TX 78249, USA

<sup>2</sup>Greehey Children’s Cancer Research Institute, University of Texas Health San Antonio, San Antonio, TX 78229, USA

<sup>3</sup>Department of Population Health Sciences, University of Texas Health San Antonio, San Antonio, TX 78229, USA

<sup>4</sup>Department of Microbiology and Molecular Genetics, University of Pittsburgh, Pittsburgh, Pennsylvania, PA 15232, USA

<sup>5</sup>Department of Medicine, School of Medicine, University of Pittsburgh, PA 15232, USA

<sup>6</sup>UPMC Hillman Cancer Center, University of Pittsburgh, PA 15232, USA





# Abstract
Since its selection as the method of the year in 2013, single-cell technologies have become mature enough to provide answers to complex research questions. However, together with the growth of single-cell profiling technologies, there has also been an increase of computational challenges to process the generated datasets. It's here that by effectively leveraging large data sets, Deep Learning (DL) is positioning as the first option for single-cell analyses. Here we provide a unified mathematical description of the DL methods used in single cell RNA sequencing (scRNA-Seq) followed with the survey of the most representative published DL algorithms for scRNA-Seq in the field.



# Key Points
* Single cell RNA sequencing technology generate large collection of transcriptomic profiles of up to millions of cells, enabling biological investigation of hidden structures or cell types, predicting their effects or responses to treatment more precisely, or utilizing subpopulation to address unanswered hypotheses. 
* Current Deep Learning-based analysis approaches for single cell RNA seq data is systematically reviewed in this paper according to the challenge they address and their roles in the analysis pipeline.
* A unified mathematical description of the surveyed DL models is presented and the specific model features were discussed when reviewing each approach.  
* A comprehensive summary of the evaluation metrics, comparision algorithms, and datasets by each approaches is presented. 




> **Keywords**
>  deep learning; single-cell RNA-seq; imputation; dimention reduction; clustering; batch correction; cell type identification; functional prediction; visualization



# 
### **Book Maintainer**
Hello, this is Sumin. Any feedback and contributions will be appreciated :)

Email: sumin.jo@utsa.edu

