
### DESC: batch correction based on clustering {#desc}

DESC [@RN88]<!--[64]--> is an AE that removes batch effect through clustering with hypothesis that batch differences in expressions are smaller than true biological variations between cell types, and, therefore, properly performing clustering on cells across multiple batches can remove batch effects without the need to define batches explicitly.  

*Model.* DESC has a conventional AE architecture. Its encoder takes normalized, log-transformed expression as the input and uses decoder output $\hat{x}_{n}$ as the reconstructed gene expression, which is equivalent to a Gaussian data distribution with $\hat{x}_{n}$ being the mean.  The loss function is
\begin{equation}
  L(\Theta) = L_{0}(\Theta)+\gamma L_{c}(\Theta) (\#eq:eq19)
\end{equation}


where $L_{0}$ is reconstruction MSE as defined in Eq. (\@ref(eq:eq8)) and $L_{c}$ is the clustering loss that regularizes the learned feature representations to form clusters. The clustering loss follows the design in the deep embedded clustering [@RN119]<!--[65]--> . Let $k \in \{1,…K\}$ be the cluster index of a cell and assume that there are $K$ clusters. Then, the clustering loss is defined as a Kullback–Leibler (KL) divergence

\begin{equation}
L_{c}(\Theta) - KL(P\|Q)  = \sum_{n=1}^{N} \sum_{k=1}^{K}p_{nk}\log{\frac{p_{nk}}{q_{nk}}} (\#eq:eq20)
\end{equation}

where $q_{nk}$ is the probability of cell n belonging to cluster $k$ and computed as

\begin{equation}
q_{nk} = \frac{ (1+\|z_{n}-\mu_{k}\|^{2})^{-1}}{\sum_{k'}(1+\|z_{n}-\mu_{k'}\|^{2})^{-1}} (\#eq:eq21)
\end{equation}


where $μ_{k}$ is the center of cluster $k$ and $p_{nk}$ is the target distribution calculated by normalizing $q_{nk}^{2}$ with frequency per cluster 

\begin{equation}
p_{nk}=\frac{q'_{nk}}{\sum_{k'=1}^{K}q'_{nk'}}, \space and \space q'_{nk}=\frac{q_{nk}^{2}}{\sum_{n=1}^{N}q_{nk}} (\#eq:eq22)
\end{equation}

	
A standard $k$-means clustering algorithm is used to initialize cluster centers. The model is first trained to minimize $L_{0}$ only to obtain the initial weights before the model is trained to optimize the combined loss (19). When the training is done, each cell's cluster index can be assigned based on $p_{nk}$. After the training, each cell is assigned with a cluster ID. 

*Evaluation metrics.* ARI was used for clustering and the KL divergence (Eq. \@ref(eq:eq20)) was used to assess the batch effect removal. 

*Results.* DESC was applied to the macaque retina dataset, which includes animal level, region level, and sample-level batch effects. The results showed that DESC is effective in removing the batch effect, whereas CCA [@RN79]<!--[29]-->, MNN [@RN84]<!--[28]-->, Seurat 3.0 [@RN96]<!--[9]-->, scVI [@RN88]<!--[13]-->, BERMUDA [@RN183]<!--[61]-->, and scanorama [@RN121]<!--[66]--> were all sensitive to batch definitions. DESC was then applied to human pancreas datasets to test its ability to remove batch effects from multiple scRNA-seq platforms and yielded the highest ARI among the comparing approaches mentioned above. When applied to human PBMC data with interferon-beta stimulation, where biological variations are compounded by batch effect, DESC was shown to be the best in removing batch effect while preserving biological variations. DESC was also shown to remove batch effect for the monocytes and mouse bone marrow data and DESC was shown to preserve the pseudotemporal structure.  Finally, DESC scales linearly with the number of cells and its running time is not affected by the increasing number of batches.











