
### scCapsNet  {#scCapsNet}

scCapsNet [@RN157]<!--[103]--> is an interpretable capsule network designed for cell type prediction. The paper showed that the trained network could be interpreted to inform marker genes and regulatory modules of cell types. 

**Model**  Two-layer architecture of scCapsNet takes log-transformed, normalized expressions as input to form a feature extraction network (consists of L parallel sigle-layer neural networks) followed by a capsule network for cell-type classification (type capsules). For each of L parallel feature extraction layer, it generates a primary capsule $\boldsymbol{u}_{l} \in \mathbb{R}^{d_{p}}$ as

\begin{equation}
\boldsymbol{u}_{l} = ReLU(\boldsymbol{W}_{P,l} \boldsymbol{x}_{n}) \forall = 1 , … , L	(\#eq:eq43)
\end{equation}

where $\boldsymbol{W}_{P,l} \in \mathbb{R}^{d_{p}×G}$ is the weight matrix. Then, the primary capsules are fed into the capsule network to compute $K$ type capsules $v_{k} \in \mathbb{R}^{d_{t}}$, one for each cell type, as

\begin{equation}
v_{k}=squash(\sum_{l}^{L}c_{kl}\boldsymbol{W}_{kl}\boldsymbol{u}_{l}) \forall k=1,…,K	(\#eq:eq44)
\end{equation}

where squash is the squashing function [@RN159]<!--[104]--> to normalize the magnitude of its input vector to be less than one, $\boldsymbol{W}_{kl}$ is another trainable weight matrix, and $c_{kl} \forall l=1,…,L$ are the coupling coefficients that represent the probability distribution of each primary capsule’s impact on the prediction of cell type $k$.  $c_{kl}$ is not trained but computed through the dynamic routing process proposed in the original capsule networks [@RN157]<!--[103]-->. The magnitude of each type of capsule $v_{k}$ represents the probability of a single cell $x_{n}$ belonging to cell type $k$, which will be used for cell type classification. 	

The training of the network minimizes the cross-entropy loss by the back-propagation algorithm. Once trained, the interpretation of marker genes and regulatory modules can be achieved by determining first the important primary capsules for each cell type and then the most significant genes for each important primary capsule (identified based on  $c_{kl}$ directly). To determine the genes that are important for an important primary capsule  $l$, genes are ranked base on the scores of the first principal component computed from the columns of $\boldsymbol{W}_{P,l}$ and then the markers are obtained by a greedy search along with the ranked list for the best classification performance. 

**Evaluation matrics** Accuracy of the predicted cell types was evaluated.   

**Results** scCapsNet’s performance was evaluated on human PBMCs [@RN161]<!--[105]--> and mouse retinal bipolar cells [@RN163]<!--[106]--> datasets and shown to have comparable accuracies (99% and 97% respectively) with DNN and other popular ML algorithms (SVM, random forest, LDA and nearest neighbor). However, the interpretability of scCapsNet was demonstrated extensively. First, examining the coupling coefficients for each cell type showed that only a few primary capsules have high values and thus is effective. Subsequently, a set of core genes were identified for each effective capsules using the greedy search on the PC-score ranked gene list. GO enrichment analysis showed that these core genes were enriched in cell type related biological functions.  Mapping the expression data into space spanned by PCs of the columns of $\boldsymbol{W}_{P,l}$ corresponding to all core genes uncovered regulatory modules that would be missed by the T-SNE of gene expressions, which demonstrated the effectiveness of the embeddings learned by scCapsNet in capturing the functionally important features.   

