
### scIGANs  {#scIGANs}

Imputation approaches based on information from cells with similar expressions suffer from oversmoothing, especially for rare cell types. scIGANs [@RN102]<!--[16]-->  is a GAN-based imputation algorithm, which overcomes this problem by using the observed samples with missing values to train a GAN to generate samples with imputed expressions. 

**Model** The gene expression vector $x_{n}$ is first reshaped into a square image-like format and scIGAN takes the reshaped data as input. The model follows a BEGAN [@RN116]<!--[60]--> framework, which substitutes the generator with an autoencoder that includes an encoder E and a decoder G and also replaces the discriminator $D$ with a function $R_{\phi_{R}}$) that computes the reconstruction error of the autoencoder (e.g. MSE).  Then, the Wasserstein distance between the reconstruction errors of the real and generated samples are computed as the loss

\begin{equation}
L(\theta,\phi) = \max_{\phi_{R}}\sum_{n=1}^{N}R_{\phi_{R}}(x_{n})-\sum_{n=1}^{N}R_{\phi_{R}}(G_{\theta}(E_{\phi}(x_{n}),y) (\#eq:eq16)
\end{equation}

The encoder and decoder are trained in a GAN fashion to minimize this Wasserstein distance. This framework forces the model to meet two computing objectives, i.e. reconstructing the real samples and discriminating between real and generated samples.  Proportional Control Theory was applied to balance these two goals during the training.  
After training, the decoder $G_{\theta}$ is used to generate new samples of a specific cell type. Then, the k-nearest neighbors (KNN) approach is applied to the real and generated samples to impute the real samples' missing expressions.  

**Evaluation metrics** It used a variety of clustering- and classification-based metrics including ARI, ACC, AUC and F-score. 

**Results** scIGANs was first tested on simulated samples with different dropout rates. Performance of rescuing the correct clusters was compared with 11 existing imputation approaches including DCA, DeepImpute, SAVER, scImpute, MAGIC, etc. scIGANs reported the best performance for all metrics. scIGAN was next evaluated for correctly clustering cell types on the Human brain scRNA-seq data and showed superior performance than existing methods again. scIGANs was next evaluated for identifying cell-cycle states using scRNA-seq datasets from mouse embryonic stem cells. The results showed that scIGANs outperformed competing existing approaches for recovering subcellular states of cell cycle dynamics. scIGANs was further shown to improve the identification of differentially expressed genes and enhance the inference of cellular trajectory using time-course scRNA-seq data from the differentiation from H1 ESC to definitive endoderm cells (DEC). Finally, scIGAN was also shown to scale to scRNA-seq methods and data sizes.  

