<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Conclusion and Discussions | Deep learning tackles single-cell analysis - A survey of deep learning for scRNA-seq analysis</title>
  <meta name="description" content="6 Conclusion and Discussions | Deep learning tackles single-cell analysis - A survey of deep learning for scRNA-seq analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Conclusion and Discussions | Deep learning tackles single-cell analysis - A survey of deep learning for scRNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="Huang-AI4Medicine-Lab/survey-of-DL-for-scRNA-seq-analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Conclusion and Discussions | Deep learning tackles single-cell analysis - A survey of deep learning for scRNA-seq analysis" />
  
  
  

<meta name="author" content="Mario Flores1ยง, Zhentao Liu1, Tinghe Zhang1, Md Musaddaqui Hasib1, Yu-Chiao Chiu2, Zhenqing Ye2,3, Karla Paniagua1, Sumin Jo1, Jianqiu Zhang1, Shou-Jiang Gao4,6, Yufang Jin1, Yidong Chen2,3ยง, and Yufei Huang5,6ยง" />


<meta name="date" content="2021-11-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ch-5.html"/>
<link rel="next" href="ch7.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Survey of Deep Learning for scRNA-seq Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About the authors</a></li>
<li class="chapter" data-level="" data-path="about-this-book.html"><a href="about-this-book.html"><i class="fa fa-check"></i>About this book</a>
<ul>
<li class="chapter" data-level="" data-path="about-this-book.html"><a href="about-this-book.html#abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="about-this-book.html"><a href="about-this-book.html#key-points"><i class="fa fa-check"></i>Key Points</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="ch-1.html"><a href="ch-1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="ch-2.html"><a href="ch-2.html"><i class="fa fa-check"></i><b>2</b> Overview of scRNA-seq processing pipeline</a></li>
<li class="chapter" data-level="3" data-path="ch-3.html"><a href="ch-3.html"><i class="fa fa-check"></i><b>3</b> Overview of common deep learning models for scRNA-seq analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="ch-3.html"><a href="ch-3.html#ch-3-1"><i class="fa fa-check"></i><b>3.1</b> Variational Autoencoder (VAEs) for scRNA-Seq data</a></li>
<li class="chapter" data-level="3.2" data-path="ch-3.html"><a href="ch-3.html#ch-3-2"><i class="fa fa-check"></i><b>3.2</b> Autoencoders (AEs) for scRNA-seq data</a></li>
<li class="chapter" data-level="3.3" data-path="ch-3.html"><a href="ch-3.html#ch-3-3"><i class="fa fa-check"></i><b>3.3</b> Generative adversarial networks (GANs) for scRNA-seq data</a></li>
<li class="chapter" data-level="3.4" data-path="ch-3.html"><a href="ch-3.html#ch-3-4"><i class="fa fa-check"></i><b>3.4</b> Supervised deep learning models</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="ch-4.html"><a href="ch-4.html"><i class="fa fa-check"></i><b>4</b> Overview of datasets and evaluation metrics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ch-4.html"><a href="ch-4.html#ch-4-1"><i class="fa fa-check"></i><b>4.1</b> Evaluation methods</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="ch-4.html"><a href="ch-4.html#ch-4-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Imputation</a></li>
<li class="chapter" data-level="4.1.2" data-path="ch-4.html"><a href="ch-4.html#ch-4-1-2"><i class="fa fa-check"></i><b>4.1.2</b> Batch effect correction</a></li>
<li class="chapter" data-level="4.1.3" data-path="ch-4.html"><a href="ch-4.html#ch-4-1-3"><i class="fa fa-check"></i><b>4.1.3</b> Clustering</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="ch-5.html"><a href="ch-5.html"><i class="fa fa-check"></i><b>5</b> Survey of deep learning models for scRNA-Seq analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-1"><i class="fa fa-check"></i><b>5.1</b> Imputation</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-1-1"><i class="fa fa-check"></i><b>5.1.1</b> DCA:deep count autoencoder</a></li>
<li class="chapter" data-level="5.1.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-1-2"><i class="fa fa-check"></i><b>5.1.2</b> SAVER-X: single-cell analysis via expression recovery harnessing external data</a></li>
<li class="chapter" data-level="5.1.3" data-path="ch-5.html"><a href="ch-5.html#ch-5-1-3"><i class="fa fa-check"></i><b>5.1.3</b> DeepImpute (Deep neural network Imputation)</a></li>
<li class="chapter" data-level="5.1.4" data-path="ch-5.html"><a href="ch-5.html#ch-5-1-4"><i class="fa fa-check"></i><b>5.1.4</b> LATE: Learning with AuToEncoder</a></li>
<li class="chapter" data-level="5.1.5" data-path="ch-5.html"><a href="ch-5.html#ch-5-1-5"><i class="fa fa-check"></i><b>5.1.5</b> scGMAI</a></li>
<li class="chapter" data-level="5.1.6" data-path="ch-5.html"><a href="ch-5.html#ch-5-1-6"><i class="fa fa-check"></i><b>5.1.6</b> scIGANs</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-2"><i class="fa fa-check"></i><b>5.2</b> Batch effect correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-2-1"><i class="fa fa-check"></i><b>5.2.1</b> BERMUDA: Batch Effect ReMoval Using Deep Autoencoders</a></li>
<li class="chapter" data-level="5.2.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-2-2"><i class="fa fa-check"></i><b>5.2.2</b> DESC: batch correction based on clustering</a></li>
<li class="chapter" data-level="5.2.3" data-path="ch-5.html"><a href="ch-5.html#ch-5-2-3"><i class="fa fa-check"></i><b>5.2.3</b> iMAP: Integration of Multiple single-cell datasets by Adversarial Paired-style transfer networks</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="ch-5.html"><a href="ch-5.html#ch-5-3"><i class="fa fa-check"></i><b>5.3</b> Dimension reduction, latent representation, clustering, and data augmentation</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-1"><i class="fa fa-check"></i><b>5.3.1</b> Dimension reduction by AEs with gene-interaction constrained architecture</a></li>
<li class="chapter" data-level="5.3.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-2"><i class="fa fa-check"></i><b>5.3.2</b> Dhaka: a VAE-based dimension reduction model</a></li>
<li class="chapter" data-level="5.3.3" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-3"><i class="fa fa-check"></i><b>5.3.3</b> cvis: a VAE for capturing low-dimensional structures</a></li>
<li class="chapter" data-level="5.3.4" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-4"><i class="fa fa-check"></i><b>5.3.4</b> scVAE: VAE for single-cell gene expression data</a></li>
<li class="chapter" data-level="5.3.5" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-5"><i class="fa fa-check"></i><b>5.3.5</b> VASC: VAE for scRNA-seq</a></li>
<li class="chapter" data-level="5.3.6" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-6"><i class="fa fa-check"></i><b>5.3.6</b> scDeepCluster</a></li>
<li class="chapter" data-level="5.3.7" data-path="ch-5.html"><a href="ch-5.html#ch-5-3-7"><i class="fa fa-check"></i><b>5.3.7</b> cscGAN: Conditional single-cell generative adversarial neural networks</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="ch-5.html"><a href="ch-5.html#ch-5-4"><i class="fa fa-check"></i><b>5.4</b> Multi-functional models</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-4-1"><i class="fa fa-check"></i><b>5.4.1</b> scVI: single-cell variational inference</a></li>
<li class="chapter" data-level="5.4.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-4-2"><i class="fa fa-check"></i><b>5.4.2</b> LDVAE: linearly decoded variational autoencoder</a></li>
<li class="chapter" data-level="5.4.3" data-path="ch-5.html"><a href="ch-5.html#ch-5-4-3"><i class="fa fa-check"></i><b>5.4.3</b> SAUCIE</a></li>
<li class="chapter" data-level="5.4.4" data-path="ch-5.html"><a href="ch-5.html#ch-5-4-4"><i class="fa fa-check"></i><b>5.4.4</b> scScope</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="ch-5.html"><a href="ch-5.html#ch-5-5"><i class="fa fa-check"></i><b>5.5</b> Doublet Classification</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-5-1"><i class="fa fa-check"></i><b>5.5.1</b> Solo</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="ch-5.html"><a href="ch-5.html#ch-5-6"><i class="fa fa-check"></i><b>5.6</b> Automated cell type identification</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-6-1"><i class="fa fa-check"></i><b>5.6.1</b> DigitalDLSorter</a></li>
<li class="chapter" data-level="5.6.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-6-2"><i class="fa fa-check"></i><b>5.6.2</b> scCapsNet</a></li>
<li class="chapter" data-level="5.6.3" data-path="ch-5.html"><a href="ch-5.html#ch-5-6-3"><i class="fa fa-check"></i><b>5.6.3</b> netAE: network-enhanced autoencoder</a></li>
<li class="chapter" data-level="5.6.4" data-path="ch-5.html"><a href="ch-5.html#ch-5-6-4"><i class="fa fa-check"></i><b>5.6.4</b> scDGN - supervised adversarial alignment of single-cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="ch-5.html"><a href="ch-5.html#ch-5-7"><i class="fa fa-check"></i><b>5.7</b> Biological function prediction</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="ch-5.html"><a href="ch-5.html#ch-5-7-1"><i class="fa fa-check"></i><b>5.7.1</b> CNNC: convolutional neural network for coexpression</a></li>
<li class="chapter" data-level="5.7.2" data-path="ch-5.html"><a href="ch-5.html#ch-5-7-2"><i class="fa fa-check"></i><b>5.7.2</b> scGen, a generative model to predict perturbation response of single cells across cell types</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ch-6.html"><a href="ch-6.html"><i class="fa fa-check"></i><b>6</b> Conclusion and Discussions</a></li>
<li class="chapter" data-level="7" data-path="ch7.html"><a href="ch7.html"><i class="fa fa-check"></i><b>7</b> Tables</a>
<ul>
<li class="chapter" data-level="7.1" data-path="ch7.html"><a href="ch7.html#deep-learning-algorithms-reviewed-in-the-paper"><i class="fa fa-check"></i><b>7.1</b> Deep Learning algorithms reviewed in the paper</a></li>
<li class="chapter" data-level="7.2" data-path="ch7.html"><a href="ch7.html#simulated-single-cell-dataalgorithms"><i class="fa fa-check"></i><b>7.2</b> Simulated single-cell data/algorithms</a></li>
<li class="chapter" data-level="7.3" data-path="ch7.html"><a href="ch7.html#human-single-cell-data-sources-used-by-different-dl-algorithms"><i class="fa fa-check"></i><b>7.3</b> Human single-cell data sources used by different DL algorithms</a></li>
<li class="chapter" data-level="7.4" data-path="ch7.html"><a href="ch7.html#mouse-single-cell-data-sources-used-by-different-dl-algorithms"><i class="fa fa-check"></i><b>7.4</b> Mouse single-cell data sources used by different DL algorithms</a></li>
<li class="chapter" data-level="7.5" data-path="ch7.html"><a href="ch7.html#single-cell-data-derived-from-other-species"><i class="fa fa-check"></i><b>7.5</b> Single-cell data derived from other species</a></li>
<li class="chapter" data-level="7.6" data-path="ch7.html"><a href="ch7.html#large-single-cell-data-source-used-by-various-algorithms"><i class="fa fa-check"></i><b>7.6</b> Large single-cell data source used by various algorithms</a></li>
<li class="chapter" data-level="7.7" data-path="ch7.html"><a href="ch7.html#evaluation-metrics-used-in-surveyed-dl-algorithms"><i class="fa fa-check"></i><b>7.7</b> Evaluation metrics used in surveyed DL algorithms</a></li>
<li class="chapter" data-level="7.8" data-path="ch7.html"><a href="ch7.html#references"><i class="fa fa-check"></i><b>7.8</b> References</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Deep learning tackles single-cell analysis - A survey of deep learning for scRNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch-6" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Conclusion and Discussions</h1>
<p>Single cell RNA-Seq technologies are tools that help exploring the cell-type composition of a particular sample and how it is dysregulated during development and disease.</p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="ch-5.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ch7.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/Huang-AI4Medicine-Lab/survey-of-DL-for-scRNA-seq-analysis/edit/main/06-conclusionAndDiscussions.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["survey-of-DL-for-scRNA-seq-analysis.pdf", "survey-of-DL-for-scRNA-seq-analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
