
### DCA:deep count autoencoder {#ch-5-1-1}

DCA [15] is an AE designed for imputation (Fig.\@ref(fig:Figure2)-B. DCA is implemented in Python as a command line and also integrated into the Scanpy framework.

*Model.* DCA models UMI counts of a cell with missing values using the ZINB distribution as

\begin{equation}
p(x_{gn} \vert \Theta)=\pi_{gn}\delta(0)+(1-\pi_{gn})NB(v_{gn},\alpha_{gn}),\space for\space g=1,…G; n=1,…N (\#eq:eq11)
\end{equation}

where $\delta(⋅$) is a Dirac delta function,$NB(,)$ denotes the negative binomial distribution and $\pi_{gn},v_{gn}, \space and \space \alpha_{gn}$ are dropout rate, mean, and dispersion, which are functions of DCA decoder output $\bf{\hat{x}_{n}}$ as

\begin{equation}
\bf{\pi_{n}}=sigmoid(\bf{W_{\pi}\hat{x}_{n}}); \bf{v_{n}}=exp(\bf{W_{v}\hat{x}_{n}}); \bf{\alpha_{n}}=exp(\bf{W_{\alpha}\hat{x}_{n}}) (\#eq:eq12)
\end{equation}

where the $g$th element of $\bf{\pi_{n}}$, $\bf{v_{n}}$, and  $\bf{\alpha_{n}}$ are $\pi_{gn}$ and $\alpha_{gn}$, respectively and  $\bf{W_{\pi}}$, $\bf{W_{v}}$, and $\bf{W_{\alpha}}$ are additional weights to be estimated. 
The DCA encoder and decoder follow the general AE formulation as in Eq. (\@ref(eq:eq7)) but the encoder takes the size factor normalized, log-transformed expression as input. 
The encoder and decoder architecture are the conventional deep neural networks. 
The parameters to be trained are $\Theta=\{\theta,\phi,W_{\pi},W_{v},W_{\alpha}\}$. 
To train the model, DCA uses a constrained log-likelihood as the loss function as

\begin{equation}
L(\Theta)=\sum_{n=1}^{N} \sum_{g=1}^{G} (-logp(x_{gn}\vert \Theta) + \lambda\pi^{2}_{gn} ) (\#eq:eq13)
\end{equation}
	
	
where the second term functions as a ridge prior on the dropout probabilities. 
Once the DCA is trained, the mean counts $v_{n}$ are used as the denoised and imputed counts for cell $n$. 

*Evaluation metrics.* A Density Resampled Estimate of Mutual Information (DREMI) measure was adapted for the higher dimensionality and sparsity of scRNA-seq data. 

*Results.* For evaluation, DCA is compared to other methods using simulated data (using Splatter R package), and a real bulk transcriptomics data from a developmental C. elegans time-course experiment was used with added simulating single-cell specific noise. For this. gene expression was measured from 206 developmentally synchronized young adults over a twelve-hour period (C elegans). Single-cell specific noise was added in silico by genewise subtracting values drawn from the exponential distribution such that 80$%$ of values were zeros.  The paper analyzed the Bulk contains less noise than single-cell transcriptomics data and can thus aid in evaluating single-cell denoising methods by providing a good ground truth model. The authors also did a comparison of other methods like SAVER [32], scImpute [33], and MAGIC[55]. DCA denoising recovered original time-course gene expression pattern while removing single-cell specific noise. Overall, DCA demonstrates the strongest recovery of the top 500 genes most strongly associated with development in the original data without noise; is shown to outperform other existing methods in capturing cell population structure in real data (using PBMC, CITE-seq, runtime scales linearly with the number of cells. 



